lib/ansible/modules/cloud/podman/podman_volume_info.py

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = """
+module: podman_volume_info
+author:
+  - "Sagi Shnaidman (@sshnaidm)"
+version_added: '2.10'
+short_description: Gather info about podman volumes
+notes: []
+description:
+  - Gather info about podman volumes with podman inspect command.
+requirements:
+  - "Podman installed on host"
+options:
+  name:
+    description:
+      - Name of the volume
+    type: str
+  executable:
+    description:
+      - Path to C(podman) executable if it is not in the C($PATH) on the
+        machine running C(podman)
+    default: 'podman'
+    type: str
+"""
+EXAMPLES = """
+- name: Gather info about all present volumes
+  podman_volume_info:
+
+- name: Gather info about specific volume
+  podman_volume_info:
+    name: specific_volume
+"""
+RETURN = """
```

Comment:
As already mentioned on IRC a couple of days ago, this documentation doesn't match the actual return values at all. The module seems to return variables `volume` (not `volumes`) and `exists` (if `name` was specified), next to `changed`, `stdout` and `stderr` which don't need to be documented). Also, here, you say `volumes` has type `dict`, but `sample` provides a `list`. I assume that if `name` is provided, the return value is a `dict` (or a one-entry `list`), and otherwise a `list`?

Reply:
Correct, I'd like to keep docker compatibility with `exists`. So if I can return both list or dict (in case of `name` given) - how can I write it here? @felixfontein can you maybe point me to an example?
Thanks

Reply:
IMO `exists` only makes sense if this module allows to only get information on one volume. If it can return information on more than one module, there'll always be problems that people interpret `exists` differently than you do. Ansible return value docs currently don't allow to have two different types for one thing.

The documentation for `podman_image_info` looks wrong. @samdoran it looks to me like it always returns a `list`, and never a `dict`?

Reply:
My idea was to keep both compatibility with docker modules, which allows only one volume to inspect and if somebody does "s/docker/podman/" it'll be same result. But to allow also to get info about all volumes, which is necessary. This case is not supported by docker, so we don't break here anything.
So if it's impossible to do, I'd choose not to implement `exists` at all.

Reply:
IMO `exists` only makes sense when the module can look at one volume. So if you want to be able to list all volumes, or filter through the list of volumes, it's better to not have `exists`, but a return value of `volumes`. (One can of course ask whether it would be better to have a module `podman_volume_info` for querying information on one volume, and `podman_volume_list` for retrieving a/the list of volumes. I don't have a good answer on that.)